credentialz/credentialz.proto

```diff
@@ -582,14 +585,36 @@ message RotateHostCredentialsResponse {
   }
 }
 
+// KeySuite specifies the valid types of keys and their lengths that
+// can be used.
+enum KeyGen {
```

Comment:
KeyGen is very similar to KeyType (line 397), should they be the same?

Reply:
Trying to keep the ssh key types and possible x.509 key types updated in different enums MAY be useful, but also introduces some accounting problems.

Reply:
I don't think that we should mix and match KeyGen and KeyType:

- KeyGen specifies, for RSA, the length to be generated as well as the key type
- There could be further divergences in the future.

From talking to [nmahabaleshwar](https://github.com/nmahabaleshwar) out of band, it sounds like there is a desire to merge the SSH and TLS (x509) key types to cut down on the number of enums. I would suggest keeping those separate. They are different protocols and I would not expect the support for the algorithms to be the same across both. Furthermore TLS likes to combine the signature algorithm with stuff like the hash type, which I think we are already proposing to include as a combined bundle.

Reply:
ok, I think I was expecting a data structure that had both the key type and the key size.
It looks like we have to know some magic numbers otherwise.

Admittedly things like ECDSA_P_521 has like 1 key size defined (which happens to be 521 bytes)...
RSA though has ~4 different key types, in this particular case you are specifying: "_2048" in the name.
So for RSA either I keep magic somewhere: "Ah, always 2048 here!" or I have to parse enum text for key lengths.

In the end Tomek seems to approve of this, so that's ok for me :) I am/was probably picking a nit that didn't need pickin'! :)